docs/reference/remap/functions/to_regex.cue

```diff
@@ -0,0 +1,37 @@
+package metadata
+
+remap: functions: to_regex: {
+	category: "Coerce"
+	description: """
+		Coerces the `value` into a regex.
+		"""
+	notices: ["Compiling a regular expression is an expensive operation and can limit Vector throughput."]
+
+	arguments: [
+		{
+			name:        "value"
+			description: "The value to convert to a regex."
+			required:    true
+			type: ["string"]
+		},
+	]
+	internal_failure_reasons: [
+		"`value` is not a string.",
+	]
+	return: {
+		types: ["regex"]
+		rules: [
+			#"If `value` is string that contains a valid regex, returns the regex constructed with this string."#,
+		]
+	}
+
+	examples: [
+		{
+			title: "Coerce to a regex"
+			source: #"""
+				to_regex("^foo$") ?? r''
+				"""#
+			return: "r'^foo$'"
```

Comment:
In the `VRL - Linux` job of the Test Suites, I'm keep having this [error](https://github.com/timberio/vector/runs/2335163206#step:4:1085) :
```
docs/functions/to_regex
  Coerce to a regex...........................................FAILED (expectation)
  "r'^foo$'"
"^foo$"
```
I suppose it's the examples here in the cue files for the docs. But if I use `vector vrl` I always get the return value `r'^foo$'`. I wonder why the produced value in the tests is a string.

Reply:
Hmm this case, it looks like the docs tests serialize the return value as JSON so it is serialized as `"^foo$"`. I could see updating them to use the `Display` representation instead (which is what you see in VRL). I pushed a commit just using the JSON serialization, for now.